### DISCUSSION TITLE
Add support for the WAI-ARIA grid pattern

### DISCUSSION THREAD

**Discussion by asabil** [chakra-ui/zag#1328] [Ideas]
# 🚀 Feature request

Add support for the WAI-ARIA grid pattern (https://www.w3.org/WAI/ARIA/apg/patterns/grid/)

## 🧱 Problem Statement / Justification

From the WAI-ARIA description:

> A [grid](https://w3c.github.io/aria/#grid) widget is a container that enables users to navigate the information or interactive elements it contains using directional navigation keys, such as arrow keys, Home, and End. As a generic container widget that offers flexible keyboard navigation, it can serve a wide variety of needs. It can be used for purposes as simple as grouping a collection of checkboxes or navigation links or as complex as creating a full-featured spreadsheet application

## ↩️ Alternatives

react-aria implements the pattern: https://react-spectrum.adobe.com/react-aria/GridList.html 

## 📝 Additional Information

* https://www.w3.org/WAI/ARIA/apg/patterns/grid/
* https://react-spectrum.adobe.com/react-aria/GridList.html

**Comment by segunadebayo** [comment]
This will land in upcoming v2
